needy/projects/autotools.py

```diff
@@ -18,19 +18,25 @@ def identifier():
 
     @staticmethod
     def is_valid_project(definition, needy):
+        failure_messages = []
         with cd(definition.directory):
             if os.path.isfile('configure'):
                 try:
                     configure_version_info = command_output(['./configure', '--version'], logging.DEBUG)
                     if 'generated by GNU Autoconf' in configure_version_info:
-                        return True
+                        return True, './configure script determined to be generated by GNU Autoconf'
                 except subprocess.CalledProcessError:
                     pass
                 except OSError:
                     pass
+                failure_message += ['./configure script was not determined to be generated by GNU Autoconf']
+            else:
+                failure_messages += ['no ./configure script found']
             if os.path.isfile('autogen.sh') and os.path.isfile('configure.ac') and os.path.isfile('Makefile.am'):
-                return True
-        return False
+                return True, 'autogen.sh, configure.ac, and Makefile.am all exist'
+            else:
+                failure_messages += ['autogen.sh, configure.ac, and Makefile.am were not all found']
```

Comment:
Generally I prefer `.append` and `.extend` over += for lists. It's just a personal preference, but if you feel like it, it would be more consistent with my existing code.